dev-utilities/release-utilities/nifi-rc-check

```diff
@@ -0,0 +1,326 @@
+if ! gpg --verify ./"nifi-$NIFI_VERSION-source-release.zip.asc" "nifi-$NIFI_VERSION-source-release.zip" ; then
+  print_error "failed to verify Nifi Assembly"
+  exit 1
+fi
+
+LOCAL_SHA1="$(sha1sum ${LOCAL_ASSEMBLY} | cut -d' ' -f1)";
+REMOTE_SHA1="$(curl -s ${NIFI_ASSEMBLY_SHA1})"
+print_section_item "Verifying Nifi Assembly SHA1"
+if [ "${LOCAL_SHA1}" = "${REMOTE_SHA1}" ] ; then
+    echo "SHA-1 match"
+else
+    print_error "SHA-1 does not match : sha1sum: ${LOCAL_SHA1} Remote sha1sum: ${REMOTE_SHA1}"
+    exit 1
+fi
+
+LOCAL_SHA256=$(shasum -a256 ${LOCAL_ASSEMBLY} | cut -d' ' -f1);
+REMOTE_SHA256=$(curl -s ${NIFI_ASSEMBLY_SHA256})
+print_section_item "Verifying Nifi Assembly SHA256"
+if [ "${LOCAL_SHA256}" = "${REMOTE_SHA256}" ] ; then
+    echo "SHA-256 match"
+else
+    echo
+    echo
+    echo sha256sum: $(shasum -a256 ${LOCAL_ASSEMBLY} | cut -d' ' -f1);
+    echo Remote sha256sum: $(curl -s ${NIFI_ASSEMBLY_SHA256})
+    print_error "SHA-256 does not match"
+    exit 1
+fi
+
+LOCAL_SHA512=$(shasum -a512 ${LOCAL_ASSEMBLY} | cut -d' ' -f1);
+REMOTE_SHA512=$(curl -s ${NIFI_ASSEMBLY_SHA512})
+print_section_item "Verifying Nifi Assembly SHA512"
+if  [ "${LOCAL_SHA512}" = "${REMOTE_SHA512}" ] ; then
+    echo "SHA-512 match"
+else
+    echo sha256sum: $(shasum -a512 ${LOCAL_ASSEMBLY} | cut -d' ' -f1);
+    echo Remote sha256sum: $(curl -s ${NIFI_ASSEMBLY_SHA512})
+    print_error "SHA-512 does not match"
+    exit 1
+fi
+
+print_section_item "Unpacking Assemblies"
+if ! unzip -q ${LOCAL_ASSEMBLY}; then
+  print_error "failed to unpack Nifi Assembly"
+  exit 1
+fi
+
+print_section_item "Retrieving remote Nifi repository"
+if ! git clone https://git-wip-us.apache.org/repos/asf/nifi.git; then
+  print_error "failed git clone https://git-wip-us.apache.org/repos/asf/nifi.git"
+  exit 1
+fi
+
+print_section_item "Getting the commit hash for ${UPPER_RC}"
+#git ls-remote --tags --refs https://github.com/apache/nifi.git | grep nifi-1.7.0-RC1 | cut -f1
+COMMIT_HASH=$(git ls-remote --tags --refs https://github.com/apache/nifi.git | grep "${NIFI_REL_RC}" | cut -f1);
+
+print_section_item "Commit Hash for ${NIFI_REL_RC} is ${COMMIT_HASH}"
+
+print_section_item "Checking out the correct tag"
+if ! git -C nifi checkout "${COMMIT_HASH}" ; then
+  print_error "failed git -C nifi checkout ${COMMIT_HASH}"
+  exit 1
+fi
+
+print_section_item "Checking for differences between downloaded rc sources and git"
+
+DIFF=$(diff --brief -r nifi-${NIFI_VERSION} nifi | grep -v DEPENDENCIES | grep -v .git* | grep -v project.basedir);
+
+if [ ! -z "${DIFF}" ] ; then
+  echo "${DIFF}"
+  print_error  "[ERROR] There are differences!"
+  exit 1
+fi
+
+print_section_header "Build and Test"
+echo ""
+echo ""
+read -p " Build Nifi and run contrib check? [yN]" -n 1 -r
+echo
+if [[ $REPLY =~ ^[Yy]$ ]]; then
+  print_section_item "Performing build with contrib-check"
+  if ! mvn -f nifi-${NIFI_VERSION}/pom.xml clean install -Pcontrib-check ; then
```

Comment:
I think this should include `-Pinclude-grpc` for verifying the build during a release vote.